### DISCUSSION TITLE
new Blocks always display Miner 0x0000...address,how to display validator address?

### DISCUSSION THREAD

**Discussion by hhq365** [0xPolygon/polygon-edge#746] [General]
## [Subject of the issue]
new Blocks always display Miner 0x0000...address,how to display validator address?

### Description
<img width="1278" alt="image" src="https://user-images.githubusercontent.com/6645976/191428998-21a911d4-71da-431e-9988-dbbb868495f1.png">

### Your environment

* OS and version:Ubuntu 20.04.5 LTS
* version of the Polygon Edge:v0.5.0
* branch that causes this issue:develop

### Steps to reproduce
1.Deploy my chain use polygon-edge develop branch, with --PoS
2.Totally deployed 4 validator node and 1 rpc node
3.Deploy blockscout manual,running mode: dev
4.Connect Blockscout to my rpc node
5.Then it displays as the screenshot

6.my blockscout env settings:
export SECRET_KEY_BASE=...
export POS_STAKING_CONTRACT="0x0000000000000000000000000000000000001001"
export ETHEREUM_JSONRPC_HTTP_URL=http://127.0.0.1:10002/
export ETHEREUM_JSONRPC_TRACE_URL=http://127.0.0.1:10002/
export ETHEREUM_JSONRPC_WS_URL=ws://127.0.0.1:10002/ws
export DATABASE_URL=...
export COIN=TEST
export COIN_NAME="TEST"
export CHAIN_ID=1006
export SUBNETWORK="TEST Testnet"
export NETWORK=""
export DISABLE_EXCHANGE_RATES="true"
export FETCH_REWARDS_WAY="manual"
export PORT=4000
export BLOCKSCOUT_VERSION=v4.1.8-beta

### Expected behaviour

Tell us what should happen
I expect to show Validator not Miner

### Actual behaviour

Tell us what happens instead
new Blocks always show “Minner 0x000“ address

### Logs
my rpc node log:
2022-09-21T06:17:45.192Z [INFO]  polygon.blockchain: new block: number=764490 txs=0 hash=0x71fda6cc34231a18230bbdfadc80651214e616895930db1b87d8f05a69322e74 parent=0xdc39f7519ca8af5698607659334af324bb06721bbc6ea6de1ec9cb7388d72d03 generation_time_in_seconds=2
2022-09-21T06:17:47.210Z [INFO]  polygon.blockchain: new block: number=764491 txs=0 hash=0xa0d8cee68913ab4b90609ffaaba91d42d04649760a6a2d8e547d68bc64180c65 parent=0x71fda6cc34231a18230bbdfadc80651214e616895930db1b87d8f05a69322e74 generation_time_in_seconds=2
2022-09-21T06:17:49.228Z [INFO]  polygon.blockchain: new block: number=764492 txs=0 hash=0xbedd615003464bef75d60b81c0be90b85325793a7d831ddc97bde299e0e83a2a parent=0xa0d8cee68913ab4b90609ffaaba91d42d04649760a6a2d8e547d68bc64180c65 generation_time_in_seconds=2
2022-09-21T06:17:51.258Z [INFO]  polygon.blockchain: new block: number=764493 txs=0 hash=0xd76ff05171678f6a24de44dd33f5ac897841d146ec28b543bac5c837cfa44042 parent=0xbedd615003464bef75d60b81c0be90b85325793a7d831ddc97bde299e0e83a2a generation_time_in_seconds=2

**Comment by ivanbozic21** [comment]
Hi @hhq365 

We will convert the issue that you opened into a discussion.

We are aware that the miner address is zero, and there is an explanation for this. 

The miner field is used only to propose a vote for a certain address, so it isn't the actual address of the proposer. 

You can find a more detailed reply in our docs [here](https://docs.polygon.technology/docs/edge/faq/validators#why-do-the-json-rpc-commands-eth_getblockbynumber-and-eth_getblockbyhash-not-return-the-miners-address).

**Comment by nnlgsakib** [comment]
> ## [Subject of the issue]
> new Blocks always display Miner 0x0000...address,how to display validator address?
> 
> ### Description
> <img alt="image" width="1278" src="https://user-images.githubusercontent.com/6645976/191428998-21a911d4-71da-431e-9988-dbbb868495f1.png">
> 
> ### Your environment
> * OS and version:Ubuntu 20.04.5 LTS
> * version of the Polygon Edge:v0.5.0
> * branch that causes this issue:develop
> 
> ### Steps to reproduce
> 1.Deploy my chain use polygon-edge develop branch, with --PoS 2.Totally deployed 4 validator node and 1 rpc node 3.Deploy blockscout manual,running mode: dev 4.Connect Blockscout to my rpc node 5.Then it displays as the screenshot
> 
> 6.my blockscout env settings: export SECRET_KEY_BASE=... export POS_STAKING_CONTRACT="0x0000000000000000000000000000000000001001" export ETHEREUM_JSONRPC_HTTP_URL=http://127.0.0.1:10002/ export ETHEREUM_JSONRPC_TRACE_URL=http://127.0.0.1:10002/ export ETHEREUM_JSONRPC_WS_URL=ws://127.0.0.1:10002/ws export DATABASE_URL=... export COIN=TEST export COIN_NAME="TEST" export CHAIN_ID=1006 export SUBNETWORK="TEST Testnet" export NETWORK="" export DISABLE_EXCHANGE_RATES="true" export FETCH_REWARDS_WAY="manual" export PORT=4000 export BLOCKSCOUT_VERSION=v4.1.8-beta
> 
> ### Expected behaviour
> Tell us what should happen I expect to show Validator not Miner
> 
> ### Actual behaviour
> Tell us what happens instead new Blocks always show “Minner 0x000“ address
> 
> ### Logs
> my rpc node log: 2022-09-21T06:17:45.192Z [INFO] polygon.blockchain: new block: number=764490 txs=0 hash=0x71fda6cc34231a18230bbdfadc80651214e616895930db1b87d8f05a69322e74 parent=0xdc39f7519ca8af5698607659334af324bb06721bbc6ea6de1ec9cb7388d72d03 generation_time_in_seconds=2 2022-09-21T06:17:47.210Z [INFO] polygon.blockchain: new block: number=764491 txs=0 hash=0xa0d8cee68913ab4b90609ffaaba91d42d04649760a6a2d8e547d68bc64180c65 parent=0x71fda6cc34231a18230bbdfadc80651214e616895930db1b87d8f05a69322e74 generation_time_in_seconds=2 2022-09-21T06:17:49.228Z [INFO] polygon.blockchain: new block: number=764492 txs=0 hash=0xbedd615003464bef75d60b81c0be90b85325793a7d831ddc97bde299e0e83a2a parent=0xa0d8cee68913ab4b90609ffaaba91d42d04649760a6a2d8e547d68bc64180c65 generation_time_in_seconds=2 2022-09-21T06:17:51.258Z [INFO] polygon.blockchain: new block: number=764493 txs=0 hash=0xd76ff05171678f6a24de44dd33f5ac897841d146ec28b543bac5c837cfa44042 parent=0xbedd615003464bef75d60b81c0be90b85325793a7d831ddc97bde299e0e83a2a generation_time_in_seconds=2

I can help you!
